### DISCUSSION TITLE
A lingering problem

### DISCUSSION THREAD

**Discussion by jgalinat** [thehighnibble/thehighnibble.github.io#64] [IMSAI8080esp replica]
Hi,
  My IMSAI8080esp has completely assembled & tested and works for the most part. I am running the latest firmware and SD image. Using the virtual TTY on the browser interface, I can boot into CP/M, examine all disks, run programs from all the disks, run the Virtual Dazzler and the Tektronix display, etc. The problem arises when I periodically get a message on the browser interface saying “System Offline. Reload?” With the line below saying  “Reloading in x seconds” .  The browser computer still shows as being connected to the IMSAI8080 host. The thing is, when this happens, the IMSAIesp continues running normally even after the browser interface reloads. Whatever operations were running during the reset are still running or have run to completion. As an example if I had a basic program running that was counting to 100,000 when the reset occurs,  and I reopen the virtual TTY console quick enough, the basic program is still running. Only the browser interface has been affected. 

Any ideas?  I have to admit, this is a really cool kit.  Already considering the S-132 add- on kit

Regards,
Jeff
  
![480FB23E-E6EA-4C0F-83CF-07893699770A](https://github.com/thehighnibble/thehighnibble.github.io/assets/131741410/2167e995-0e2d-40b5-b102-bb0f8b70a332)

**Comment by kcaine** [comment]
If it helps, I have experienced this often also.

**Reply by jgalinat** [reply]
Actually, that does help quite a bit!  Just to know that I am not the only
one experiencing this means a great deal!
Thank you!

**Comment by agn453** [comment]
Might be Wi-Fi signal strength/channel interference causing this.  The esp32 needs a strong Wi-Fi signal.  Try moving your IMSAI8080esp to a different location - or change the Wi-Fi setup to use a different channel (choose the best from the 2.4GHz channel 1, 6 or 11 as these do not overlap).

Tony

**Reply by jgalinat** [reply]
Hi Tony,  
  Thanks for the reply!  Signal strength shouldn’t be a problem as the 8080esp is on the same table as my IMAC.   I am not sure about signal interference so I will give changing the WiFi channel a try.  Please forgive my ignorance but how do I change the WiFi channel on the ESP32? I assume I have to edit something in the boot sequence?  I have not connected the ESP32  to my home network yet so I am still just connecting to the default imsai8080 access point. Can you give me any pointers?
 Regards,
Jeff

**Reply by agn453** [reply]
I recommend you switch to connecting to your home network (the imsai8080 may be using the same channel as your normal Wi-Fi access point/router/gateway device).  Sorry I don’t know if it is possible to manually force the channel used by the esp32.

**Reply by jgalinat** [reply]
I will put the 8080esp on my home network today and see if the situation improves. Thanks again for your suggestions!

**Comment by udo-munk** [comment]
This is normal behavior, same would happen if you disconnect a terminal from the serial port of some computer running a CP/M program.

**Reply by jgalinat** [reply]
Hi Udo,
  I only used the example of the running BASIC program to illustrate that the ESP32 itself was not resetting & I was just losing the WiFi connection. Why I keep losing the WiFi connection is the big problem.

**Comment by nbreeden2** [comment]
If you have two tabs or browsers open to the IMSAI it can do this. One will take control, then the other and it will ping-pong back and forth.

**Comment by jgalinat** [comment]
Hi Neil
  Thanks for responding. Unfortunately, this happens when I only have a
single browser tab open. I tried to connect the 8080esp to my home network
but it didn’t work. I will outline the steps I took in another post so
hopefully someone can point out where I went wrong.
Regards,
Jeff
